nuxeo-runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/model/DefaultComponent.java

```diff
@@ -67,10 +75,24 @@ public void unregisterExtension(Extension extension) {
         setModifiedNow();
     }
 
-    public void registerContribution(Object contribution, String extensionPoint, ComponentInstance contributor) {
+    public void registerContribution(Object contribution, String extensionPoint, ComponentInstance component) {
+        if (contribution instanceof Descriptor) {
+            Descriptor descriptor = (Descriptor) contribution;
+            String target = getTarget(component, extensionPoint);
+            registry.register(target, extensionPoint, descriptor);
+            getLog().debug(String.format("Registered %s to %s.%s", descriptor.getId(), target, extensionPoint));
+            contributionsChanged(descriptor);
+        }
     }
 
-    public void unregisterContribution(Object contribution, String extensionPoint, ComponentInstance contributor) {
+    public void unregisterContribution(Object contribution, String extensionPoint, ComponentInstance component) {
+        if (contribution instanceof Descriptor) {
+            Descriptor descriptor = (Descriptor) contribution;
+            String target = getTarget(component, extensionPoint);
```

Comment:
I'm thinking about replacing this search by an String getComponentId() in DefaultComponent that would be called there, wdyt ?

Reply:
Adding getComponentId() I will just remove the first argument that would be useless as-is.